src/txn/txn_ckpt.c

```diff
@@ -604,11 +604,9 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
          * because recovery doesn't set the recovery timestamp until its checkpoint is complete.
          */
         if (txn_global->has_stable_timestamp) {
-            txn->read_timestamp = txn_global->stable_timestamp;
-            txn_global->checkpoint_timestamp = txn->read_timestamp;
-            F_SET(txn, WT_TXN_HAS_TS_READ);
+            txn_global->checkpoint_timestamp = txn_global->stable_timestamp;
             if (!F_ISSET(conn, WT_CONN_RECOVERING))
-                txn_global->meta_ckpt_timestamp = txn->read_timestamp;
+                txn_global->meta_ckpt_timestamp = txn_global->stable_timestamp;
```

Comment:
could we use txn_global->checkpoint_timestamp instead of txn_global->stable_timestamp ?

Reply:
It won't change anything functionally as we're holding the txn_global->rwlock but I've updated it.